vapi/appliance/simulator/simulator.go

```diff
@@ -194,32 +131,24 @@ func (h *Handler) sshStatus(writer http.ResponseWriter, request *http.Request) {
 	}
 }
 
-func (h *Handler) shellStatus(writer http.ResponseWriter, request *http.Request) {
-	update := func(enabled bool, timeout int) {
-		h.shell.Lock()
-		fmt.Println("Hello")
-		h.shell.access.Enabled = enabled
-		h.shell.access.Timeout = timeout
-		h.shell.Unlock()
+func (h *Handler) shellAccess(writer http.ResponseWriter, request *http.Request) {
+	delayReset := func(timeout int) {
+		time.AfterFunc(time.Duration(timeout)*time.Second, func() {
+			h.shell.Enabled = false
+			h.shell.Timeout = 0
+		})
 	}
 
 	switch request.Method {
 	case http.MethodGet:
-		h.shell.RLock()
-		vapi.StatusOK(writer, h.shell.access)
-		h.shell.RUnlock()
+		vapi.StatusOK(writer, h.shell)
+
 	case http.MethodPut:
 		var input shell.Access
 		if h.decode(request, writer, &input) {
-			if h.shell.t != nil {
-				h.shell.t.Stop()
-			}
-			update(input.Enabled, input.Timeout)
-			h.shell.t.Reset(time.Duration(input.Timeout) * time.Second)
-			go func() {
-				<-h.shell.t.C
-				update(false, 0)
-			}()
+			h.shell.Enabled = input.Enabled
+			h.shell.Timeout = input.Timeout
+			go delayReset(input.Timeout)
```

Comment:
@dougm since this code runs asynchronously, it would not be protected by the global lock and cause races IMHO?